src/app/loginComponents/accounts/external/accounts.service.ts

```diff
@@ -76,4 +76,12 @@ export class AccountsService {
         break;
     }
   }
+
+  /**
+   * Compute the root portion of a URL.  Adapted from:
+   * {@link https://stackoverflow.com/questions/1368264/how-to-extract-the-hostname-portion-of-a-url-in-javascript}
+   */
+  getRoot(url: string): string {
+    return url.replace(/^(.*\/\/[^\/?#]*).*$/,"$1");
```

Comment:
3rd option of the link seems cleaner. if you're worried about IE11, we already have a polyfill for it

Reply:
Yeah, I went with the regexp solution vs URL because it seemed more resistant to browser oddities, although more opaque.